packages/meditrak-app/app/entityMenu/EntityList.js

```diff
@@ -96,29 +105,29 @@ export class EntityList extends PureComponent {
       return;
     }
 
-    const { entities } = this.props;
-    const lowerCaseSearchTerm = searchTerm.toLowerCase();
-
-    const searchResults = entities
-      .filter(
-        ({ name, parentName }) =>
-          name.toLowerCase().includes(lowerCaseSearchTerm) ||
-          (parentName && parentName.toLowerCase().includes(lowerCaseSearchTerm)),
-      )
-      .sort(({ name: a }, { name: b }) => {
-        // Send entity names that start with the search term to the top.
-        const checkMatchesAtStart = name => name.toLowerCase().startsWith(lowerCaseSearchTerm);
-        if (checkMatchesAtStart(a) !== checkMatchesAtStart(b)) {
-          return checkMatchesAtStart(a) ? -1 : 1;
-        }
+    const { entities } = this.state;
 
-        // Sort alphabetically
-        return a.localeCompare(b);
+    const primarySearchResults = entities.filtered(`name BEGINSWITH[c] $0`, searchTerm);
+    if (primarySearchResults.length >= ENOUGH_SEARCH_RESULTS) {
+      this.setState({
+        searchTerm,
+        searchResults: primarySearchResults,
       });
-    this.setState({
-      searchTerm,
-      searchResults,
-    });
+    } else {
+      // if there aren't enough search results for the user to look through, try including
+      // matches later in the name, or matches on the parent entity
+      const secondarySearchResults = entities.filtered(
+        `(NOT name BEGINSWITH[c] $0) AND (name CONTAINS[c] $0 OR parent.name BEGINSWITH[c] $0) LIMIT(${
```

Comment:
```suggestion
        `(NOT name BEGINSWITH[c] $0) AND (name CONTAINS[c] $0 OR parent.name CONTAINS[c] $0) LIMIT(${
```
To be consistent with previous behaviour?

Reply:
Also, what happens here if parent is `NULL`?

Reply:
I didn't mention this decision anywhere but think the old logic gave parents way too much weight, and this makes it more like "if you really want to filter by the parent, you can bloody well type their name properly, and even then they'll only come up as secondary results!"
I should check that decision with a product owner though (which used to be me for meditrak ;-))

Reply:
Parent is null - I'll check!

Reply:
Tested the parent is null case and it's fine :-)

Reply:
And followed up on the UX of filtering on parent [here](https://linear.app/bes/issue/RN-466#comment-2eb4bcea)